### DISCUSSION TITLE
latex pdf features - output latex tables, global code settings

### DISCUSSION THREAD

**Discussion by actsasgeek** [quarto-dev/quarto-cli#1457] [Q&A]
I just learned about Quarto recently and I'm out kicking the tires. I publish several sets of course notes from Jupyter Notebooks as PDF (currently using jupyterbook, former nbsphinx user). 

There are two questions/issues I've run into and perhaps they have a solution.

## Code Generated Table Display in PDF

I find that I'm running into several of the same problems I have experienced with other publishing systems. It is very, very convenient to build a table in code...but nearly impossible to output one properly.

### using raw latex 

raw latex works in a Markdown cell but not in code output:

```
import tabulate
print(tabulate.tabulate({"a": [1, 2, 3], "b": ["$\\beta_1$", 2, 3]}, tablefmt="latex_raw"))
```

will output

```
\begin{tabular}{rl}
\hline
 1 & $\beta_1$ \\
 2 & 2         \\
 3 & 3         \\
\hline
\end{tabular}
```

but it just gets directly printed in the PDF.

### Using display_latex

```
from IPython.display import display_latex
import tabulate
display_latex(tabulate.tabulate({"a": [1, 2, 3], "b": ["$\\beta_1$", 2, 3]}, tablefmt="latex_raw"))
```

doesn't work either. In fact, the output is completely omitted from the PDF.

### Markdown

Outputting markdown instead doesn't work and Markdown does not support Math mode.

### Use a Pandas DataFrame

Somehow, Pandas DataFrame's do get rendered correctly. I would guess that there's some _repr_latex_ magic at wor. I would think that display_latex would set the same field in the json. The issue here, as with markdown, is that embedded math isn't properly rendered:

```
import pandas as pd

df = pd.DataFrame({"a": [1, 2, 3], "b": ["$\\beta_1$", 2, 3]})
df
```

and DataFrames are a poor substitute for a properly formatted table.

I would sort of expect some kind of code cell level directive like:

```
code-output-format: latex
```
where the values might be html, latex, markdown, and default to raw

## Global Code Cell Declarations

I was sort of surprised that code-overflow and some other code cell declarations aren't global. It seems like one would want to be able to set code overflow once rather than for every cell. But maybe I did it wrong...

```
---
title: "My document"
format:
  pdf:
    toc: true
    number-sections: true
    colorlinks: true
    code-block-border-left: false
    code-overflow: wrap
---

One final note. I often find that PDF output gets short shrift because the idea seems to be that HTML is the way to go (it is probably also easier). And I'm very happy to see that this doesn't appear to be the case for Quarto. 

For example, that you output PDF for Mermaid is amazing. 

Just one vote but my students hate HTML textbooks. They can't refer to page numbers. They can make notes in them like they can with PDFs and some readers. They can't print them out well or easily or reliably. They like to just put the PDF on their phone or tablet and read it on the subway and not worry about internet connection.

**Comment by cderv** [comment]
I did not tested yet but it seems you want to output raw result from you python code chunk. 
Did you settign the `output: asis` chunk option https://quarto.org/docs/reference/cells/cells-jupyter.html#cell-output 

Without setting this, the output of the chunk will be including in special code output container and not consider as raw text (raw LaTeX or raw Markdown) 

Just a hint

**Reply by actsasgeek** [reply]
Thank you! That fixed that. I think I took this statement a bit too literally, "Treat output as raw **markdown** with no enclosing containers."

Do you have to set the wrapped code for each code cell or did I miss something?

**Reply by cderv** [reply]
You mean setting `output: asis`  option ? Do you need to do that on each code cell or can you do that globally ? 
not sure I correctly understood the question. Sorry

**Reply by actsasgeek** [reply]
no worries.

It appears to me that I need to set code-overflow: wrap on each code cell. Is there a global setting that I'm missing? It tried to set it at the start of the document and it didn't do anything.

**Reply by cderv** [reply]
> It tried to set it at the start of the document and it didn't do anything.

Can you share example ? I think it should work
